modules/tinymce/src/themes/silver/test/ts/browser/editor/core/ChoiceControlsTest.ts

```diff
@@ -0,0 +1,303 @@
+import { UiFinder, Waiter } from '@ephox/agar';
+import { beforeEach, context, describe, it } from '@ephox/bedrock-client';
+import { Arr, Optional } from '@ephox/katamari';
+import { TinyAssertions, TinyHooks, TinySelections, TinyUiActions } from '@ephox/mcagar';
+import { Attribute } from '@ephox/sugar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+import Theme from 'tinymce/themes/silver/Theme';
+
+interface ToolbarOrMenuSpec {
+  readonly name: string;
+  readonly pOpen: (editor: Editor, title: string) => Promise<void>;
+  readonly close: (editor: Editor, title: string) => void;
+  readonly menuSelector: string;
+}
+
+describe('browser.tinymce.themes.silver.editor.core.ChoiceControlsTest', () => {
```

Comment:
This is a pretty messy test now... I think it would be better to have 2 separate test files for line height and content language. If you're really against that, then we should have a `context` per type instead to properly abstract the 2 different types of tests happening here and the prefixes on each separate context should be removed.

Reply:
Because so much of the logic is in the choice menu abstraction, rather than the specific implementations of line height or language buttons, I'd rather test it this way. I've added the contexts as you've asked